src/core/agent/services/signifyNotificationService.ts

```diff
@@ -31,27 +32,19 @@ class SignifyNotificationService extends AgentService {
       nextIndex: 0,
       lastNotificationId: "",
     };
-    try {
-      notificationQuery = (await PreferencesStorage.get(
-        PreferencesKeys.APP_KERIA_NOTIFICATION_MARKER
-      )) as unknown as KeriaNotificationMarker;
-    } catch (error) {
-      if (
-        (error as Error).message ==
-        `${PreferencesStorage.KEY_NOT_FOUND} ${PreferencesKeys.APP_KERIA_NOTIFICATION_MARKER}`
-      ) {
-        // Set the preference key
-        await PreferencesStorage.set(
-          PreferencesKeys.APP_KERIA_NOTIFICATION_MARKER,
-          {
-            nextIndex: 0,
-            lastNotificationId: "",
-          }
-        );
-      } else {
-        throw error;
-      }
-    }
+    const notificationQueryRecord = await Agent.agent.basicStorage.findById(
+      MiscRecordId.APP_KERIA_NOTIFICATION_MARKER
+    );
+    if (!notificationQueryRecord) {
+      await createOrUpdateBasicRecord(
```

Comment:
We already called `findById` above this so we can just use `basicStorage.save` directly